src/datepicker/datepicker.ts

```diff
@@ -293,10 +309,9 @@ export class NgbDatepicker implements OnDestroy,
   onTouched = () => {};
 
   constructor(
-      private _service: NgbDatepickerService, private _calendar: NgbCalendar, public i18n: NgbDatepickerI18n,
-      config: NgbDatepickerConfig, private _keyboardService: NgbDatepickerKeyboardService, cd: ChangeDetectorRef,
-      private _elementRef: ElementRef<HTMLElement>, private _ngbDateAdapter: NgbDateAdapter<any>,
-      private _ngZone: NgZone) {
+      public _service: NgbDatepickerService, private readonly _calendar: NgbCalendar, public i18n: NgbDatepickerI18n,
```

Comment:
- `_service` → no reason to make it public, should stay as before
- `cd` → no reason to make it private, should stay as before